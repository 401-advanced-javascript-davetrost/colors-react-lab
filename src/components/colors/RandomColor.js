import React, { Component } from 'react';
import colors from '../../data/commonColorNames';
import ColorCircle from './ColorCircle';

export default class RandomColor extends Component {

  state = {
    backgroundColor: 'black',
    backgroundImage: ''
  };

  updateColor = color => {
    if(color === this.state.backgroundColor) {
      this.setState({
        backgroundColor: 'transparent',
        backgroundImage: 'url(https://i.dailymail.co.uk/i/pix/2016/03/18/15/324D202500000578-3498922-image-a-33_1458315465874.jpg)' 
      });
    } else {
      this.setState({
        backgroundColor: color,
        backgroundImage: '' 
      });
    }
  }

  componentDidMount() {
    setInterval(
      () => {
        const color = colors[Math.floor(Math.random() * colors.length)];
        this.updateColor(color);
      },
      1000
    );
  }

  render() {
    return (
      <ColorCircle
        backgroundColor={this.state.backgroundColor}
        backgroundImage={this.state.backgroundImage} 
      />
    );
  }
}
