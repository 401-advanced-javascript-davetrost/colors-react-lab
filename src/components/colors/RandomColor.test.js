import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('Random Color component', () => {
  it('renders a ColorCircle component', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
