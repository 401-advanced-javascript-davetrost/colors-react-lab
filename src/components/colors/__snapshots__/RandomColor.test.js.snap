// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random Color component renders a ColorCircle component 1`] = `
<ColorCircle
  backgroundColor="black"
  backgroundImage=""
/>
`;
